sandbox/main_test.go

```diff
@@ -0,0 +1,194 @@
+		{{- end }}
+			{{- if .ResponseBody }}
+			Body:
+				 {{ .ResponseBody }}
+			{{- end }}
+		<== END HTTP
+{{- end }}
+=== END ===`))
+
+type Log struct {
+	Method         string
+	URL            string
+	RequestHeader  http.Header
+	RequestBody    string
+	Duration       time.Duration
+	Status         string
+	ResponseHeader http.Header
+	ResponseBody   string
+	Error          string
+}
+
+type LoggingHTTPClient struct {
+	next zooz.HTTPClient
+	t    *testing.T
+}
+
+func (c LoggingHTTPClient) Do(request *http.Request) (*http.Response, error) {
+	log := Log{
+		Method:        request.Method,
+		URL:           request.URL.String(),
+		RequestHeader: request.Header,
+	}
+	defer func() {
+		builder := &strings.Builder{}
+		err := LogTemplate.Execute(builder, log)
+		assert.NoError(c.t, err)
+		c.t.Log(builder.String())
+	}()
+
+	if request.Body != nil {
+		reqBody, err := io.ReadAll(request.Body)
+		require.NoError(c.t, err)
+		require.NoError(c.t, request.Body.Close())
+		log.RequestBody = string(reqBody)
+		request.Body = io.NopCloser(strings.NewReader(string(reqBody)))
+	}
+
+	start := time.Now()
+	response, err := c.next.Do(request)
+	log.Duration = time.Since(start)
+	if err != nil {
+		log.Error = fmt.Sprintf("%+v", err)
+		return nil, err
+	}
+
+	log.Status = response.Status
+	log.ResponseHeader = response.Header
+
+	if response.Body != nil {
+		respBody, err := io.ReadAll(response.Body)
+		require.NoError(c.t, err)
+		require.NoError(c.t, response.Body.Close())
+		log.ResponseBody = string(respBody)
+		response.Body = io.NopCloser(strings.NewReader(string(respBody)))
+	}
+
+	return response, nil
+}
+
+func GetClient(t *testing.T) *zooz.Client {
+	var errs []string
+
+	appID := os.Getenv(EnvAppID)
+	if appID == "" {
+		errs = append(errs, "missing env "+EnvAppID)
+	}
+	privateKey := os.Getenv(EnvPrivateKey)
+	if privateKey == "" {
+		errs = append(errs, "missing env "+EnvPrivateKey)
+	}
+	publicKey := os.Getenv(EnvPublicKey)
+	if publicKey == "" {
+		errs = append(errs, "missing env "+EnvPublicKey)
+	}
+
+	require.Empty(t, errs)
+
+	var httpClient zooz.HTTPClient = http.DefaultClient
+	if os.Getenv(EnvLogRequests) != "" {
+		httpClient = LoggingHTTPClient{
+			next: httpClient,
+			t:    t,
+		}
+	}
+
+	return zooz.New(
+		zooz.OptHTTPClient(httpClient),
+		zooz.OptAppID(appID),
+		zooz.OptEnv(zooz.EnvTest),
+		zooz.OptPrivateKey(privateKey),
+		zooz.OptPublicKey(publicKey),
+	)
+}
+
+func randomString(l int) string {
+	bytes := make([]byte, l)
+	for i := 0; i < l; i++ {
+		bytes[i] = randChar()
+	}
+	return string(bytes)
+}
+
+// randChar returns random character in range 0-9 or A-Z or a-z
+func randChar() byte {
+	r := rand.Intn(10 + 26 + 26)
+	if r < 10 {
+		// 48 - 57 -> digits
+		return byte(48 + r)
+	}
+
+	r -= 10
+	if r < 26 {
+		// 65 - 90 -> capital letters
+		return byte(65 + r)
+	}
+
+	r -= 26
+	// 97 - 122 -> letters
+	return byte(97 + r)
+}
+
+func softAssert(t *testing.T, do func()) {
+	do()
+	if t.Failed() {
+		t.FailNow()
```

Comment:
I took it from java world (name and the concept).

I thought about `requireAll`. But rejected it because you should not use `require` package in `do` func. And this may be strange:
```go
require.NoError(t, err)
requireAll(t, func() {
	assert.Equal(t, ...)
	assert.Equal(t, ...)
	assert.Equal(t, ...)
})
```

I understand why `softAssert` in go is strange, but can't find better name.

Before introduction of the function I used following code
```go
require.NoError(t, err)
assert.Equal(t, ...)
assert.Equal(t, ...)
assert.Equal(t, ...)
if t.Failed() {
	return
}
```

Maybe it's better to go back and use that approach?

Reply:
Good name, thanks.
Also added doc string.